src/Maker/MakeEvent.php

```diff
@@ -23,42 +23,78 @@
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Question\Question;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\KernelEvents;
 
 /**
  * @author Javier Eguiluz <javier.eguiluz@gmail.com>
  * @author Ryan Weaver <weaverryan@gmail.com>
+ * @author Steven Renaux <steven.renaux8000@gmail.com>
  */
-final class MakeSubscriber extends AbstractMaker
+final class MakeEvent extends AbstractMaker
 {
-    public function __construct(private EventRegistry $eventRegistry)
+    private const ALL_TYPES = ['Subscriber', 'Listener'];
+    private bool $isSubscriber = false;
+
+    public function __construct(private readonly EventRegistry $eventRegistry)
     {
     }
 
     public static function getCommandName(): string
     {
-        return 'make:subscriber';
+        return 'make:event';
```

Comment:
I don't know because for me it's also sounds like disturbing to me to write `make:subscriber` to get a listener at the end.

Maybe let see what the others reviewers will think about the naming. It's not written in stone :wink:

Reply:
IMO, a subscriber is no more than a listener which brings its configuration (event => method). `make:event-listener --as-subscriber` could be an alternative.

Reply:
It might be too complex, but here is a suggestion :
With `make:event-listener` as the command name, we could have some logic that checks :
- If there's a `--as-subscriber` or `--as-listener` option. If so, we stick with that.
- Else, try to parse the classname's suffix. If we can determine what type of class the user wants to generate, stick with that.
- Else, ask a question, with `EventListener` as default

Reply:
Yea, if we call it `make:event-listener`... you can still run it with `make:event` :). But I probably like `make:listener` best. We think of "event listeners" in Symfony, JavaScript, etc. "listener" is a dominate name that has a lot of meaning. 

> If there's a --as-subscriber or --as-listener option. If so, we stick with that.

This is a nice idea, but I don't think we need it. I'm cool with looking at the suffix of that class and, if it's not `Listener` or `Subscriber`, asking them which they want. Or, maybe better, just using a listener, as they are so nice nowadays thanks to `AsEventListener`.